parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java

```diff
@@ -403,15 +405,29 @@ public ParquetWriter(Path file, Configuration conf, WriteSupport<T> writeSupport
 
     this.codecFactory = codecFactory;
     CompressionCodecFactory.BytesInputCompressor compressor = codecFactory.getCompressor(compressionCodecName);
+
+    final Map<String, String> extraMetadata;
+    if (encodingProps.getExtraMetaData() == null
+        || encodingProps.getExtraMetaData().isEmpty()) {
+      extraMetadata = writeContext.getExtraMetaData();
+    } else {
+      extraMetadata = new HashMap<>(writeContext.getExtraMetaData());
+
+      encodingProps.getExtraMetaData().forEach((metadataKey, metadataValue) -> {
+        if (metadataKey.equals(OBJECT_MODEL_NAME_PROP)) {
```

Comment:
Can we avoid specializing any key? IIUC, it can also be caught at line 422 if OBJECT_MODEL_NAME_PROP has been set already.

Reply:
This key is kind of a special case since it's added by the delegated `InternalParqueRecordWriter` at the end of writing: https://github.com/apache/parquet-mr/blob/945836c/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordWriter.java#L132-L134

So if we remove this extra check and there's a conflicting `OBJECT_MODEL_NAME_PROP` key in `extraMetaData`, `InternalParquetRecordWriter` will silently overwrite it at the end.